indy-utils/src/main/kotlin/com/luxoft/blockchainlab/hyperledger/indy/IndyUser.kt

```diff
@@ -1,596 +1,202 @@
 package com.luxoft.blockchainlab.hyperledger.indy
 
-import com.luxoft.blockchainlab.hyperledger.indy.helpers.TailsHelper
+import com.luxoft.blockchainlab.hyperledger.indy.ledger.LedgerService
 import com.luxoft.blockchainlab.hyperledger.indy.models.*
-import com.luxoft.blockchainlab.hyperledger.indy.roles.IndyIssuer
-import com.luxoft.blockchainlab.hyperledger.indy.roles.IndyProver
-import com.luxoft.blockchainlab.hyperledger.indy.roles.IndyTrustee
-import com.luxoft.blockchainlab.hyperledger.indy.roles.IndyVerifier
 import com.luxoft.blockchainlab.hyperledger.indy.utils.SerializationUtils
-import com.luxoft.blockchainlab.hyperledger.indy.utils.getRootCause
-import mu.KotlinLogging
-import org.hyperledger.indy.sdk.anoncreds.Anoncreds
-import org.hyperledger.indy.sdk.anoncreds.DuplicateMasterSecretNameException
+import com.luxoft.blockchainlab.hyperledger.indy.wallet.WalletService
 import org.hyperledger.indy.sdk.did.Did
-import org.hyperledger.indy.sdk.pairwise.Pairwise
-import org.hyperledger.indy.sdk.pool.Pool
-import org.hyperledger.indy.sdk.wallet.Wallet
-import org.hyperledger.indy.sdk.wallet.WalletItemNotFoundException
-import java.util.concurrent.ExecutionException
 
 
 /**
  * The central class that encapsulates Indy SDK calls and keeps the corresponding state.
- *
- * Create one instance per each server node that deals with Indy Ledger.
+ * This is implementation of [IndyFacade] so it should support every valid [LedgerService] and [WalletService]
+ *  implementation.
  */
-open class IndyUser(
-    val pool: Pool,
-    val wallet: Wallet,
-    did: String?,
```

Comment:
By the way, did we previously had control over our DID? and could switch to other identities?

Reply:
Unfortunately no. We have same poor control over our DID as before.